pkg/volume/host_path/host_path.go

```diff
@@ -99,14 +100,22 @@ func (plugin *hostPathPlugin) GetAccessModes() []v1.PersistentVolumeAccessMode {
 	}
 }
 
-func (plugin *hostPathPlugin) NewMounter(spec *volume.Spec, pod *v1.Pod, _ volume.VolumeOptions) (volume.Mounter, error) {
+func (plugin *hostPathPlugin) NewMounter(spec *volume.Spec, pod *v1.Pod, opts volume.VolumeOptions) (volume.Mounter, error) {
 	hostPathVolumeSource, readOnly, err := getVolumeSource(spec)
 	if err != nil {
 		return nil, err
 	}
 
+	path := hostPathVolumeSource.Path
+	// A containerized kubelet would have difficulty creating directories.
+	// The implementation will likely respect the containerized flag, allowing it to be "/rootfs/" aware and
+	// thus operate as desired.
+	if opts.Containerized {
+		path = filepath.Join("/rootfs", path)
```

Comment:
@lucab @euank I tried to use `nsenter` for containerized kubelet. But it has some limitations.

1. First, we have to use `--privileged` to give extended privileges to kubelet container;
2. Also we have to get a pid on the host, either using `--pid=host` or mounting an existing proc to the container;
3. For binary exec file `nsenter`, it may not exist on all container environments, such as [`ubuntu:trusty`](https://hub.docker.com/_/ubuntu/). I think depending on `nsenter` is fragile.

Also cc @thockin @luxas